cmd/isp/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility isp`
+//!
+//! The LPC55 has a built-in In-System Programming (ISP) mode. ISP mode
+//! does _not_ work over the regular debug probe but over UART. You must
+//! specificy the UART path specifcially
+//!
+//! $ humility isp --port /dev/ttyUSB0 read-memory 0x0 32
+//!             \/  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
+//! 0x00000000 | 00 40 00 20 31 01 00 00 f3 02 00 00 a5 04 00 00 | .@. 1...........
+//! 0x00000010 | a1 01 00 00 a3 01 00 00 a5 01 00 00 a7 01 00 00 | ................
+//!
+
+use anyhow::Result;
+use byteorder::ByteOrder;
+use clap::Command as ClapCommand;
+use clap::{CommandFactory, Parser};
+use humility::hubris::*;
+use humility_cmd::{Archive, Args, Command, Dumper};
+use serialport::{DataBits, FlowControl, Parity, SerialPortSettings, StopBits};
+use std::io::Read;
+use std::path::PathBuf;
+
+mod cmd;
+mod protocol;
+
+use crate::protocol::BootloaderProperty;
+
+#[derive(Parser, Debug)]
+#[clap(name = "subcmd")]
+enum IspCmd {
+    /// Runs a single ping to verify communication with the target
+    Ping,
+    /// Reads memory from the specified address
+    ReadMemory {
+        #[structopt(parse(try_from_str = parse_int::parse))]
```

Comment:
Nit: now that we're on Clap 3, you can use `clap(parse(...))` instead of `structopt(parse(...))` for all of these.